packages/ocvImpact/R/utils_targeting.R

```diff
@@ -104,102 +113,128 @@ load_targets_by_country <- function(datapath, modelpath, country){
-  coverage_as_all_0_for_campaign <- (sum(coverage$coverage) == 0)
   if (!coverage_as_all_0_for_campaign){
+    
     coverage <- import_coverage_scenario(modelpath, country, scenario, filter0 = TRUE, redownload = FALSE)
+    
   }
-
+  
   if (is.null(coverage)){
-
+    
     message("No vaccine was assigned for the no-vaccination scenario. Returning NULL value.")
     ftargets_flat <- NULL
-  
+    
   } else{
-
+    
     ## Perform checks on the coverage scenario
     if (!all(coverage$gender == "both") | 
         !all(coverage$age_range_verbatim == "default age and gender" | coverage$age_range_verbatim == ">1y" | coverage$age_range_verbatim == "default age groups" | coverage$age_range_verbatim == "<NA>" | coverage$age_range_verbatim == "1-100")|
         !all(coverage$activity_type == "campaign")
-        ){
+    ){
       stop(paste("Vaccine assignment is not supported for this coverage scenario. Check the gender, age_range_verbatim, and activity_type columns in the", scenario, "coverage sheet."))
     }
-
+    
     ftargets <- vector(mode = "list", length = length(coverage$year))
-
+    
     for (i in 1:length(coverage$year)){
-
+      
       cov_year <- coverage[i,]
-      goal_target_pop <- cov_year$target
-      goal_fvp <- cov_year$fvp
+      
+      ##calam commented out on 1/23/2024 because it does not apply to implementation for 202310gavi-4 touchstone
+      ##goal_target_pop <- cov_year$target
+      
+      ##goal fvps with one dose and two doses of the vaccine
+      goal_ocv1_fvp <- cov_year$fvp_ocv1
+      goal_ocv2_fvp <- cov_year$fvp_ocv2
       
       ## Skip locations vaccinated within the previous `num_skip_years`
       skip_years <- cov_year$year - (1:num_skip_years)
       if(num_skip_years>0 & length(skip_years)>0 & any(skip_years %in% coverage$year)){
-
+        
         message(paste("Skip locations vaccinated in", paste(skip_years, collapse = ",")))    
         incl_skip_years <- which(names(ftargets) %in% skip_years)    
         recent_targets <- unique(as.vector(sapply(incl_skip_years, function(x){
           ftargets[[x]]$GID_2
         })))
-
+        
         new_ptargets <- dplyr::filter(ptargets, !(GID_2 %in% recent_targets))      
-
+        
       } else{
         message(paste("Skipping was disabled (num_skip_years=0) or there were no recent campaigns within", num_skip_years, "years of", cov_year$year))
         new_ptargets <- ptargets
       }
-
+      
       ## full district targeting 
       ptargets_avail <- run_targeting_strategy(new_ptargets, targeting_strat) %>%
         dplyr::mutate(
           id = seq_along(GID_2),
-          possible_fvp = round(goal_target_pop * pop_prop * campaign_cov, 0), 
-          possible_fvp_cumsum = cumsum(possible_fvp),
-          actual_fvp = ifelse(possible_fvp_cumsum <= goal_fvp, possible_fvp, 0),
-          fullDistrict_target = ifelse(actual_fvp > 0, TRUE, FALSE))
-
+          possible_fvp_ocv1 = round(goal_ocv1_fvp * pop_prop * campaign_cov, 0),
```

Comment:
This is where the problem is. We need to multiply by the target population, not goal_ocv1_fvp here...